src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -0,0 +1,153 @@
+#include "CIESProfile.h"
+
+using namespace nbl;
+using namespace asset;
+
+const double& CIESProfile::getIntegral() const 
+{
+    size_t numHSamples = 2 * getHoriAngles().size();
+    size_t numVSamples = 2 * getVertAngles().size();
+    double dTheta = core::PI<double>() / numVSamples;
+    double dPhi = 2 * core::PI<double>() / numHSamples;
+    double dThetaInAngle = MAX_VANGLE / numVSamples;
+    double dPhiInAngle = MAX_HANGLE / numHSamples;
+    double res = 0;
+    for (size_t i = 0; i < numVSamples; i++) {
+        for (size_t j = 0; j < numHSamples; j++) {
+            res += dPhi * dTheta * sin(dTheta * i) * sample(i * dThetaInAngle, j * dPhiInAngle);
+        }
+    }
```

Comment:
compute straight from the original irregular grid, no resampling

Reply:
done, but we have 0.089475341451795226 error, I need to think of it

Reply:
that error is probably because the sampled thing was obviously inaccurate